source/slang/slang-ir-metal-legalize.cpp

```diff
@@ -1284,9 +1284,131 @@ namespace Slang
                     varLayoutBuilder.findOrAddResourceInfo(LayoutResourceKind::MetalPayload);
                     auto paramVarLayout = varLayoutBuilder.build();
                     builder.addLayoutDecoration(param, paramVarLayout);
+
```

Comment:
It is cleaner to put all these logic into a separate function called legalizeMeshEntryPoint, and called from legalizeEntryPoint based on the stage in a switch statement so we can add other stage specific legalizations easily in the future.